core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -398,164 +410,185 @@ public SyncStatus sync() throws AccessControlException, InvalidPathException {
       return SyncStatus.NOT_NEEDED;
     }
     Instant startTime = Instant.now();
-    try (LockedInodePath path =
-             mInodeTree.lockInodePath(mRootScheme, mRpcContext.getJournalContext())) {
-      if (mAuditContext != null && mAuditContextSrcInodeFunc != null) {
-        mAuditContext.setSrcInode(mAuditContextSrcInodeFunc.apply(path));
-      }
+    try (LockedInodePath path = lockSyncRoot()) {
+      try {
+        if (mAvoidConcurrentSync) {
+          // In case that current ts = sync start ts
+          long concurrentBlockerSyncInterval = Instant.now().toEpochMilli() - syncStartTime + 1;
+          // TODO(elega) if a previous concurrent sync failed, the sync will trigger again
+          // This can be fixed by adding a lastSyncAttemptTs in the UfsSyncPathCache for entries
+          // and let InodeSyncStream report failed metadata sync event.
+          boolean shouldSync = mUfsSyncPathCache.shouldSyncPath(mRootScheme.getPath().toString(),
+              concurrentBlockerSyncInterval, mIsGetFileInfo).isShouldSync();
+          if (!shouldSync) {
+            return SyncStatus.NOT_NEEDED;
+          }
+          if (mRootScheme.getPattern() == LockPattern.READ) {
+            path.downgradeToRead();
+          }
```

Comment:
If it's load metadata only, the lock is acquired in the READ mode.
https://github.com/Alluxio/alluxio/blob/master/core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java#L674-L678
So we need to downgrade it to read after the should sync check finishes